src/cc65/main.c

```diff
@@ -951,6 +962,9 @@ int main (int argc, char* argv[])
 
                 case 'O':
                     IS_Set (&Optimize, 1);
+                    IS_Set (&CodeSizeFactor, 100);
+                    IS_Set (&EnableRegVars, 0);
+                    IS_Set (&InlineStdFuncs, 0);
```

Comment:
I don't see -O2 and -O3 etc. different from -Osi or -Osir. Internally the gcc level also just enable or disable different optimization steps.

If cc65 would behave like gcc it would be rather simple to change the Makefile to opit specific optimizations for "problem files".

But I noticed that cc65 has a --disable-opt switch. This could be used in this example without changing the existing behaviour of the -O switches.